tests/10-client-api.spec.common.cjs

```diff
@@ -31,6 +31,43 @@ describe('http-client API', () => {
     err.requestUrl.should.equal('http://httpbin.org/status/404');
     err.response.status.should.equal(404);
   });
+  it('handles a connection refused error', async () => {
+    let err;
+    let response;
+    // the intention here is to use an used http port
+    // the port used can not be higher than 65535 making it illegal
+    const nonExistentResource = 'http://localhost:65535';
+    const expectedErrorCode = 'ECONNREFUSED';
+    try {
+      response = await httpClient.get(nonExistentResource);
+    } catch(e) {
+      err = e;
+    }
+    should.not.exist(
+      response, 'Expected nonExistentResource to not return a response.');
+    should.exist(
+      err, 'Expected nonExistentResource to error.');
+    should.not.exist(
+      err.response,
+      'Expected nonExistentResource "err.response" to not exist.'
+    );
+    should.exist(
+      err.requestUrl,
+      'Expected nonExistentResource "err.requestUrl" to exist.'
+    );
+    err.requestUrl.should.equal(
+      nonExistentResource,
+      `Expected nonExistentResource "err.requestUrl" to be ` +
+        `${nonExistentResource}`
+    );
+    // node 18's global fetch seems to be changing the error return type
```

Comment:
"seems to"?  please use a more factual comment that says what is happening in node 18 vs others.

Reply:
I changed the language here: https://github.com/digitalbazaar/http-client/pull/29/commits/f7bd52d5eadc92cc5ea9c512b6b9caf1b2665c41 but the weakness of the assertion accurately represents the amount of research done. Node 18 introduced global fetch, it turns out: https://github.com/nodejs/undici/blob/c7f527b2a727a726d9e53c19297314d2850717cb/lib/fetch/response.js#L104-L107

it takes the original error and assigns it to `cause`. I'm still not 100% sure that is why in node 18 we start getting `code` in `cause`, but it seems likely as the reason.